components/ws-manager/pkg/manager/ws_daemon_controller.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
```

Comment:
Can tests be added? I ask because I'm unsure due to usage of `sigs.k8s.io/controller-runtime/pkg/client`.